samplePDF/samplePDFFDBase.cpp

```diff
@@ -100,105 +253,77 @@ void samplePDFFDBase::SetupSampleBinning(){
   }
-  else if(XVarStr.length() > 0 && YVarStr.length() > 0){
-	MACH3LOG_INFO("Setting Up 2D binning");
-	MACH3LOG_INFO("{} : {}", XVarStr, YVarStr);  
-	set2DBinning(SampleXBins, SampleYBins);
+  MACH3LOG_INFO("YBinning:");
+  MACH3LOG_INFO("{}", YBinEdgesStr);
+  
+  //Check whether you are setting up 1D or 2D binning
+  if(nDimensions == 1){
+    MACH3LOG_INFO("Setting up 1D binning with {}", XVarStr);
+    set1DBinning(SampleXBins);  
   }
-}
-
-void samplePDFFDBase::UseBinnedOscReweighting(bool ans) 
-{
-  osc_binned = ans;
-  if(ans == true) {
-    std::cout << "WARNING: you are using binned oscillation weight without specifying the binning. It will use E_reco binning to set the energy (recommended to use set1Dbinning first !). If you want another binning use : \n useBinnedOscReweighting(true, int, double*) \nwhere int is the number of bins and double* is an array with the bins boundaries." << std::endl ;
-
-    /// Get the binning from the MC histogram
-    const int nb_bins = _hPDF1D -> GetXaxis() -> GetNbins() ;      
-    const double* osc_bins = _hPDF1D -> GetXaxis() -> GetXbins() -> GetArray();
-    osc_binned_axis = new TAxis(nb_bins, osc_bins) ;
+  
+  else if(nDimensions == 2){
+    MACH3LOG_INFO("Setting up 2D binning with {} and {}", XVarStr, YVarStr);
+    set2DBinning(SampleXBins, SampleYBins);
   }
-
-  return;
-}
-
-void samplePDFFDBase::UseBinnedOscReweighting(bool ans, int nbins, double *osc_bins) 
-{
-  osc_binned = ans;
-  if(ans == true) {
-    osc_binned_axis = new TAxis(nbins, osc_bins) ;
+  else{
+    MACH3LOG_ERROR("Number of dimensions is not 1 or 2, this is unsupported at the moment");
+    throw MaCh3Exception(__FILE__, __LINE__);
   }
-
-  return;
+  
+  return; 
 }
 
 // ************************************************
 bool samplePDFFDBase::IsEventSelected(const int iSample, const int iEvent) {
-// ************************************************
-
+  // ************************************************
+  
   double Val;
-
-  for (unsigned int iSelection=0;iSelection < Selection.size() ;iSelection++) {
-    
+  for (unsigned int iSelection=0;iSelection < Selection.size() ;iSelection++) {  
     Val = ReturnKinematicParameter(Selection[iSelection][0], iSample, iEvent);
-    //std::cout << "Val returned for selection " << iSelection << " is " << Val << std::endl;
-    //DB If multiple return values, it will consider each value seperately
-    //DB Already checked that Selection vector is correctly sized
-    
-    //DB In the case where Selection[0].size()==3, Only Events with Val >= Selection[iSelection][1] and Val < Selection[iSelection][2] are considered Passed
     if ((Val<Selection[iSelection][1])||(Val>=Selection[iSelection][2])) {
       return false;
     }
-
   }
-
-  //DB To avoid unneccessary checks, now return false rather than setting bool to true and continuing to check
+  
+  //DB To avoid unnecessary checks, now return false rather than setting bool to true and continuing to check
```

Comment:
the comment says that it will return false, but it doesn't, is it in the wrong place or outdated?